src/app/entry/router-link.pipe.ts

```diff
@@ -1,25 +1,71 @@
 import { Pipe, PipeTransform } from '@angular/core';
 import { EntryType } from '../shared/enum/entry-type';
-import { Workflow } from '../shared/openapi';
+import { CollectionEntry, Workflow, DockstoreTool } from '../shared/openapi';
 
 @Pipe({
   name: 'routerLink',
 })
 export class RouterLinkPipe implements PipeTransform {
-  transform(entryType: EntryType, workflow: Workflow): string | null {
-    if (!workflow || !entryType) {
+  /**
+   * Returns the complete path for a workflow, tool, or collection entry
+   * To be used for routerLink navigation
+   *
+   * @param entryType
+   * @param entry
+   * @returns string | null
+   */
+  transform(entryType: EntryType, entry: DockstoreTool | Workflow | CollectionEntry): string | null {
+    if (!entry || !entryType) {
       return null;
     }
-    switch (entryType) {
-      case EntryType.BioWorkflow:
-        return '/workflows/' + workflow.full_workflow_path;
-      case EntryType.Service:
-        return '/services/' + workflow.full_workflow_path;
-      case EntryType.AppTool:
-        return '/containers/' + workflow.full_workflow_path;
-      case EntryType.Notebook:
-        return '/notebooks/' + workflow.full_workflow_path;
+
+    if ((entry as Workflow).full_workflow_path) {
```

Comment:
Steve mentioned that we now have `EntryTypeMetadata` in this [comment](https://github.com/dockstore/dockstore-ui2/pull/1756#discussion_r1171580682) and I think using that here would greatly simplify this function. `EntryTypeMetadata.sitePath` contains the entry type specific portion of the site path (ex: `containers`), so that would remove the need for all the switches.

`EntryTypeMetadata` should be a field in `DockstoreTool | Workflow`, so you can easily access it there. As for `CollectionEntry`, it's not a field in there, but you can use this function in the `EntryTypeMetadataService` to retrieve the `EntryTypeMetadata` based on the collection entry's `entryType`: https://github.com/dockstore/dockstore-ui2/blob/3478bfb2e97481e1b127224bdb06b268ef8c94a1/src/app/entry/type-metadata/entry-type-metadata.service.ts#L30